backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/shopware6/interceptor/MD_Available_For_Sales.java

```diff
@@ -39,19 +39,19 @@
 {
 	private final ITrxManager trxManager = Services.get(ITrxManager.class);
 
-	private final ExportStockToShopwareExternalSystem exportStockToShopwareExternalSystem;
+	private final ExportAvailableForSalesToShopwareExternalSystem exportAvailableForSalesToShopwareExternalSystem;
 
-	public MD_Available_For_Sales(@NonNull final ExportStockToShopwareExternalSystem exportStockToShopwareExternalSystem)
+	public MD_Available_For_Sales(@NonNull final ExportAvailableForSalesToShopwareExternalSystem exportAvailableForSalesToShopwareExternalSystem)
 	{
-		this.exportStockToShopwareExternalSystem = exportStockToShopwareExternalSystem;
+		this.exportAvailableForSalesToShopwareExternalSystem = exportAvailableForSalesToShopwareExternalSystem;
 	}
 
 	@ModelChange( //
-			timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE })
-	public void exportStockToShopware(@NonNull final I_MD_Available_For_Sales availableForSales)
+			timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE, ModelValidator.TYPE_AFTER_DELETE })
+	public void exportAvailableForSalesToShopware(@NonNull final I_MD_Available_For_Sales availableForSales)
```

Comment:
i guess this relateds to Lars's note 
> When the QtyOnHandStock reaches 0, the shopware availableStock won't be updated. The record in MD_Available_For_Sales was removed.

but are we sure this even works though? did you test it and does availableForSales.getM_Product_ID() actually return a result after delete? if not, you might need to register your trx-listener on TYPE_BEFORE_DELETE..
if there is no cucumber-testabout putting the stock to 0, please create one

Reply:
Yes, there is a cucumber test about putting the stock to 0. I added the scenario to the `UAT Test Documentation` in the `Creating and changing shipments` section, test no 500 and annotated the cucumber scenario with the respective id.